.github/workflows/integration.yaml

```diff
@@ -0,0 +1,81 @@
+name: Tests
+
+on:
+  pull_request:
+  schedule:
+    - cron: '0 8 * * TUE'
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: pip install tox
+
+    - name: Lint code
+      run: tox -vve lint
+
+  unit-test:
+    name: Unit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: pip install tox
+
+      - name: Run unit tests
+        run: tox -e unit
+
+      # Upload coverage report on failure
+      - name: Upload coverage artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: coverage
+          path: training-operator-cov_html
+        if: failure()
+
+  integration-test:
+    name: Integration
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+            provider: microk8s
+            channel: 1.21/stable
+
+      # https://github.com/charmed-kubernetes/actions-operator/issues/27
+      - name: Install charmcraft latest/candidate
```

Comment:
Stylistic choice, but for steps that simply run a single command, I prefer not adding a `name` field, as the default name is pretty good, and it's one more place for code and comments to get out of sync. As an example, if we skipped the `name` field on this, it would show up as `Run sudo snap refresh charmcraft --channel latest/candidate`.

Reply:
+1, though I wanted to left it very evident so we could remove it later. Now that we have [PR#13](https://github.com/charmed-kubernetes/actions-operator/pull/28) I will remove this block.